Fw/Buffer/test/ut/TestBuffer.cpp

```diff
@@ -83,21 +83,19 @@ void test_representations() {
     }
     Fw::SerializeStatus stat = sbb.serialize(100);
     ASSERT_NE(stat, Fw::FW_SERIALIZE_OK);
+
     // And that another call to repr resets it
-    sbb = buffer.getSerializeRepr();
```

Comment:
I think that we will want to support this.  We should probably change the return type from `getSerializeRepr` to return the `ExternalBufferWithMemberCopy` for the same reason.  `Fw::Buffer` owns the "slice" and thus we want member copy in these instances.

Reply:
What is the intended purpose of the deleted line here? I suspect that what it is actually doing is not intended:

1 The right-hand side gets a reference to the serialize representation object, expressed as an Fw::SerializeBufferBase.
1. The left-hand side sbb is a reference to the same object, from a previous call to getSerializeRepr, also expressed as an Fw::SerializeBufferBase.
1. According to the copy semantics prior to this PR, assignment of an Fw::SerializeBufferBase copies the data. So I believe the deleted line copies the same data on top of itself, i.e., it is a no-op.
1. In the revised copy semantics, the assignment is not allowed. I think this is the correct behavior, since assigning to a base-class reference is contrary to C++ coding recommendations.

With regard to the internal implementation of the serialize representation, we could use ExternalBufferWithMemberCopy, and that would provide member copy semantics, as in the new version of DpContainer. However, I think this is not necessary. Fw::Buffer already defines its own copy semantics that does the same thing (member copy) in handwritten code.

Reply:
Also, I don't think we want to return an ExternalBufferWithMemberCopy. This would be dangerous, because then users could assign into the buffer and update the external buffer pointer. If we returned ExternalBufferWithDataCopy then we could preserve what was happening in the old code (assignment that copies data) but I don't see a need to assign into the serial buffer here. It seems to be that all access through the serial buffer should be through the serialization interface, not through assignment.

Reply:
So I think the current return type Fw::SerializeBufferBase is correct.

Reply:
I see.  Ok, I am fine with this as-is.